core/contract/kernel/kernel.go

```diff
@@ -62,6 +63,42 @@ var (
 	ErrInvalidChainName = errors.New("Invalid Chain name")
 )
 
+type KernelMethodFunc func(desc *contract.TxDesc) error
+
+type kernelFunc map[string]KernelMethodFunc
+
+func (k *Kernel) getRunKernelFunc() kernelFunc {
+	return kernelFunc{
```

Comment:
This map object should not be created every time it is called. By the way, it doesn't need to create the kernelFunc type, just use the map type as the return value.

Reply:
Done :)